android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java

```diff
@@ -131,28 +131,38 @@ public void setupDialog(@NonNull Dialog dialog, int style) {
         networkName.setText(mNetworkName);
         ImageView icon = (ImageView) view.findViewById(R.id.account_picture);
         icon.setImageResource(mAccountPic);
+        String valueToConvert = mTxInfo.txData.baseData.value;
```

Comment:
I think you need to check `mTxInfo.txType` here  for `ERC20Transfer` as you can have other contract call transaction types from dapps or perhaps from future functionality we add.

```
enum TransactionType {
  ETHSend,
  ERC20Transfer,
  ERC20Approve,
  Other
};
```